File: Chapter 1 Naming Conventions/GuessTheNumber.py
import random
def isValidGuess(userGuess):
    if userGuess.isdigit() and 1<= int(userGuess) <=100:
        return True
    else:
        return False

def main():
    numberToGuess=random.randint(1,100)
    isCorrectGuess=False
    userGuess=input("Guess a number between 1 and 100:")
    guessCount=0
    while not isCorrectGuess:
        if not isValidGuess(userGuess):
            userGuess=input("I wont count this one Please enter a number between 1 to 100")
            continue
        else:
            guessCount+=1
            userGuess=int(userGuess)

        if userGuess<numberToGuess:
            userGuess=input("Too low. Guess again")
        elif userGuess>numberToGuess:
            userGuess=input("Too High. Guess again")
        else:
            print("You guessed it in",guessCount,"guesses!")
            isCorrectGuess=True

File: Chapter 2 Functions/Assignment 1/src/GuessTheNumber.java
import java.util.Random;
import java.util.Scanner;

public class GuessTheNumber {

    private final int numberToGuess;
    private int guessCount;
    private final Scanner scanner;

    public GuessTheNumber() {
        this.numberToGuess = generateRandomNumber();
        this.guessCount = 0;
        this.scanner = new Scanner(System.in);
    }

    private int generateRandomNumber() {
        return new Random().nextInt(101);
    }

    private int getUserGuess() {
        while (true) {
            System.out.print("Guess a number between 1 and 100: ");
            String userInput = scanner.nextLine();
            if (isValidGuess(userInput)) {
                return Integer.parseInt(userInput);
            } else {
                System.out.println("Your input is Invalid. Enter a number between 1 and 100.");
            }
        }
    }

    private boolean isValidGuess(String userInput) {
        int number = Integer.parseInt(userInput);
        return number >= 1 && number <= 100;
    }

    private void evaluateGuess(int userGuess) {
        if (userGuess < numberToGuess) {
            System.out.println("Too low! Try again.");
        } else if (userGuess > numberToGuess) {
            System.out.println("Too high! Try again.");
        } else {
            System.out.println("Your guess is correct. guess count = " + guessCount);
        }
    }

    public void GuessNumber() {
        boolean isCorrectGuess = false;
        while (!isCorrectGuess) {
            int userGuess = getUserGuess();
            guessCount++;
            if (userGuess == numberToGuess) {
                isCorrectGuess = true;
            }
            evaluateGuess(userGuess);
        }
        scanner.close();
    }

    public static void main(String[] args) {
        GuessTheNumber game = new GuessTheNumber();
        game.GuessNumber();
    }
}